docs/breaking-changes.md

```diff
@@ -0,0 +1,44 @@
+# Version upgrades and breaking changes
+
+Evolving the Lockbook code base requires some special considerations. 
+
+## Breaking change between core versions
+
+Because we support offline use our clients store and maintain local state. If we update our app, we need to make sure the new version of the app can gracefully interact with the current data on the device. Ultimately we can always nuke the local state and re-sync with the server. We want to avoid doing this where possible for the following reasons:
+
++ We would need to explicitly keep track of offline changes that haven't yet been synced
++ Depending on the user this could take a long time
++ Needlessly places a load our infrastructure
+
+Where possible therefore, where possible we should support old functionality. When this is not possible (a data format changed), a "migration" must be performed. Encoded in rust forever will be instructions on how to step through each version of state. These migrations should not require an internet connection. All core endpoints that interact with state may return `MigrationRequired`, you can also explicitly check the status by calling `get_db_state`.
```

Comment:
> Where possible therefore, where possible

Can correct to `Therefore, where possible, ...` or just `Where possible, ...`

> ... return MigrationRequired, you can...

Should be `, and you can ...` or a new sentence `. You can ...` or a semicolon `; you can ...`